src/Units/CodeContextProvider/CodeContextProvider.vala

```diff
@@ -54,83 +54,80 @@ namespace Units {
         return -1;
       }
 
-      Vala.CodeContext context_internal = new Vala.CodeContext();
-      Vala.CodeContext.push (context_internal);
+      context = new Vala.CodeContext();
```

Comment:
I think that with that change, code parsing in a separate thread will not work (that's why I added the context_internal in the first place). We need that though, as parsing could take several seconds while locking the UI.
If you find a better solution, please notify me!